### ISSUE TITLE
replacing DipatchQueue.main with background queue

### ISSUE THREAD

**Issue by murad1981** [neekeetab/CachingPlayerItem#21]
the following call `asset.resourceLoader.setDelegate` in each `init` method, should use background queue `DispatchQueue.global(qos: .background)` instead of main queue to prevent possible app freezes on the main thread which happened with me.

**Comment by ppamorim** [comment]
`DispatchQueue..global(qos: .background)` is too low and weak. Should be at least `DispatchQueue.global(qos: .default)`.

**Comment by neekeetab** [comment]
I agree that the work of `ResourceLoaderDelegate` should be done on a global queue. At the same time, the delegate calls should be performed on the main queue.

**Comment by ppamorim** [comment]
@neekeetab I've changed to `DispatchQueue.global(qos: .default)` and it's working fine.

**Comment by zackperdue** [comment]
@neekeetab I try this and mine still crashes. Any more info on this?
